packages/typescript-estree/src/node-utils.ts

```diff
@@ -191,6 +191,7 @@ export function isComment(node: ts.Node): boolean {
  * @returns is JSDoc comment
  */
 function isJSDocComment(node: ts.Node): node is ts.JSDoc {
+  // eslint-disable-next-line deprecation/deprecation -- SyntaxKind.JSDoc was only added in TS4.7 so we can't use it yet
```

Comment:
this is the compromise to removing the patch.
you can't declaration merge a comment onto an enum member.
small price to pay though.